### PR TITLE
feat: switch from Monaco to CodeMirror

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Ticket: WEB-1241

The Monaco editor that we have been using, although is a great tool, it is very heavy in size... The proposal here is to replace it with CodeMirror (using the react wrapper).

## Test Plan

Check if the functionality of all instances of where we used Monaco still works:

- Code tab (under experiments)
- modal view filters
- experiment create modal
- jupyter lab modal
- experiment configuration
- markdown 
- workspace create modal

Screen Shot 2023-05-22 at 16 25 38
Screen Shot 2023-05-22 at 16 25 24
Screen Shot 2023-05-22 at 16 25 16
Screen Shot 2023-05-22 at 16 25 07
Screen Shot 2023-05-24 at 15 36 19
Screen Shot 2023-05-24 at 15 34 47

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/kit/CodeEditor.tsx [hunk 1]
```diff
@@ -249,18 +249,11 @@ const CodeEditor: React.FC<Props> = ({ files, onSelectFile, readonly, selectedFi
...
-          language={getSyntaxHighlight()}
-          options={{
-            minimap: {
-              enabled: false,
-            },
-            occurrencesHighlight: false,
-            readOnly: readonly,
-            showFoldingControls: 'always',
-          }}
+          readOnly
```
Comment (mapmeld) [inline webui/react/src/components/kit/CodeEditor.tsx:255]: should be `readOnly={readonly}`

File: webui/react/src/components/WorkspaceCreateModal.tsx [hunk 2]
```diff
@@ -193,14 +193,7 @@ const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }
                       },
                     },
                   ]}>
-                  <MonacoEditor
-                    height="16vh"
-                    options={{
-                      readOnly: !canModifyCPS,
```
Comment (mapmeld) [inline webui/react/src/components/WorkspaceCreateModal.tsx:199]: there are a few places like this where the old editor had `readOnly` and should be passed to `CodeMirrorEditor`

**Review by ashtonG — CHANGES_REQUESTED** [review]
noticed we're only seeing .4MB in savings on the vendor bundle with this PR -- given the tests in the ERD, it should be more. Looking at the bundle analysis, it seems like the `@uiw/codemirror-extensions-langs` package is importing every language package. We should only import the languages we need.

**Comment by johnkim-det** [comment]
> > is there a reason why `CodeEditor` is in the UI kit and imports `CodeMirrorEditor`, but the app uses `CodeMirrorEditor` instead of `CodeEditor`?
> > ideally the UI kit component would not import any components outside of the UI kit, and the app would use the UI kit component.
> 
> AFAIK, `CodeEditor`, so far, is the component for the `Code` tab 🤔, but, I guess we could open a ticket for replacing that into the UI kit

Oh I see, didn't realize the difference between the two, but that makes sense. I suppose a ticket for adding `CodeMirrorEditor` to the UI kit at some point could be good.

Until that change happens, will still probably want to make a copy of `CodeMirrorEditor` within the `kit` folder, per https://hpe-aiatscale.atlassian.net/browse/WEB-1243. But I can just do that in my PR if you merge this one first.

**Review by ashtonG — CHANGES_REQUESTED** [review]
We are still importing every language module, This time via `@codemirror/language-data`. Do we need it for code highlighting in markdown?

**Review by ashtonG — APPROVED** [review]
image
let's goooooo

**Pull request merged**